OAK-10340: Child order not updated on source after move

mreutegg opened the pull request.

mreutegg committed:
OAK-10340: Child order not updated on source after move

Test to reproduce the issue

mreutegg committed:
OAK-10340: Child order not updated on source after move

Ensure child order on old and new parent

stefan-egli commented on oak-core/src/main/java/org/apache/jackrabbit/oak/core/MutableTree.java:
```diff
@@ -264,11 +264,14 @@ void setParentAndName(@NotNull MutableTree parent, @NotNull String name) {
         boolean success = nodeBuilder.moveTo(newParent.nodeBuilder, newName);
         if (success) {
-            parent.updateChildOrder(false);
+            name = newName;
+            parent = newParent;
```
+1 to fixing this one as well - even though it doesn't fix anything, since the MutableTree (source) is thrown away upon returning from `move` - but it was wrong so far, so good to fix this anyway.

mreutegg replied:
Right. I forgot to mention this. I'll add a note to the JIRA issue.

mreutegg replied:
And I'm going to add a test for this.

mreutegg replied:
Done with https://github.com/apache/jackrabbit-oak/pull/1017/commits/f23db0f19ea64d82f7fb771467ba75626f63aaf7

stefan-egli reviewed (approved):
+1, lgtm

mreutegg committed:
OAK-10340: Child order not updated on source after move

Additional test that verifies Tree parent stays the same when move fails at NodeBuider.moveTo()

mreutegg committed:
Merge branch 'trunk' into OAK-10340
